Made the build instructions more detailed (#4087)
ppy/osu: README.md

- A desktop platform with the [.NET Core SDK 2.2](https://www.microsoft.com/net/learn/get-started) or higher installed.
- When working with the codebase, we recommend using an IDE with intellisense and syntax highlighting, such as [Visual Studio Community Edition](https://www.visualstudio.com/) (Windows), [Visual Studio Code](https://code.visualstudio.com/) (with the C# plugin installed) or [Jetbrains Rider](https://www.jetbrains.com/rider/) (commercial).

# Running osu!

## Releases

If you are not interested in developing the game, please head over to the [releases](https://github.com/ppy/osu/releases) to download a precompiled build with automatic updating enabled.

> - Windows 64 bit users should download and run `install.exe`.

Reviewer: ```suggestion
- Windows (x64) users should download and run `install.exe`.
```

> - MacOS users should download and run `osu.app.zip`.

Reviewer: ```suggestion
- macOS users (10.12 "Sierra" and higher) should download and run `osu.app.zip`.
```

> There is currently no release for Windows 32 bit, Linux, or any other platform. If you are not running Windows 64 bit or MacOS, you should build osu! from the source code (see below).

Reviewer: ```suggestion
If your platform is not listed above, there is still a chance you can manually build it by following the instructions below.
```

> There is currently no release for Windows 32 bit, Linux, or any other platform. If you are not running Windows 64 bit or MacOS, you should build osu! from the source code (see bellow).

Reviewer 2: ```suggestion
There is currently no release for Windows 32 bit, Linux, or any other platform. If you are not running Windows 64 bit or MacOS, you should build osu! from the source code (see below).
```
"bellow" should be below

- A desktop platform with the [.NET Core SDK 2.2](https://www.microsoft.com/net/learn/get-started) or higher installed.
- When working with the codebase, we recommend using an IDE with intellisense and syntax highlighting, such as [Visual Studio Community Edition](https://www.visualstudio.com/) (Windows), [Visual Studio Code](https://code.visualstudio.com/) (with the C# plugin installed) or [Jetbrains Rider](https://www.jetbrains.com/rider/) (commercial).

# Running osu!

## Releases

If you are not interested in developing the game, please head over to the [releases](https://github.com/ppy/osu/releases) to download a precompiled build with automatic updating enabled.

- Windows 64 bit users should download and run `install.exe`.

> - MacOS users should download and run `osu.app.zip`.

Reviewer 3: macOS should have a lower case "m".

- iOS users can join the [TestFlight beta program](https://t.co/xQJmHkfC18).

> There is currently no release for Windows 32 bit, Linux, or any other platform. If you are not running Windows 64 bit or MacOS, you should build osu! from the source code (see below).

Reviewer 3: This is poorly worded and makes it sound as if there are no releases whatsoever.

## Downloading the source code

Clone the repository **including submodules**:

```
git clone --recurse-submodules https://github.com/ppy/osu
cd osu
```

> If you forgot the `--recurse-submodules` option, run this command inside the `osu` directory:
>
> `git submodule update --init --recursive`

To update the source code to the latest commit, run the following command inside the `osu` directory:

```
git pull --recurse-submodules
```

## Building

> Configuration for Visual Studio 2017, Rider and Visual Studio Code is included in the source code.

Reviewer: ```suggestion
Build configurations for [Visual Studio 2017+](https://visualstudio.microsoft.com/vs/), [Jetbrains Rider](https://www.jetbrains.com/rider/) and [Visual Studio Code](https://code.visualstudio.com/) are included in the source code.
```

> Visual Studio Code users must run the `Restore` task before any build attempt.

> You can also build osu! from the command-line, with `dotnet`:

Reviewer: ```suggestion
You can also build osu! from the command-line:
```

```

> dotnet restore

Reviewer: not necessary (please double-check)

dotnet run --project osu.Desktop -c Release
```

> The `-c Release` option **must be omitted** when building for development purposes.

Reviewer: i am VERY hesitant to word this readme towards people trying to get performance rather than actually debug the project. Maybe inverse this note or remove altogether (and remove all release build arguments).

### A note for Linux users

> On Linux, the environment variable `LD_LIBRARY_PATH` must point to the build directory (replace `Release` with `Debug` in the following paths if you do not want to include the `-c Release` flag).

Reviewer: as above

> The build directory is located at `osu.Desktop/bin/Release/netcoreappX.X`, where `X.X` is the version of the .NET Core SDK (see Requirements). The required value can be found in `osu.Desktop/osu.Desktop.csproj`, look for a line starting with "TargetFramework".

Reviewer: too verbose.

For example, you can run osu! with the following commands:

```

> export NETCORE_VERSION="$(grep TargetFramework osu.Desktop/osu.Desktop.csproj | sed -r 's/.*>(.*)<\/.*/\1/')"

Reviewer: inline in command rather than exporting
